src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvokerProvider.cs

```diff
@@ -171,15 +176,36 @@ private InnerCache CurrentCache
                 modelReleaser = _modelFactoryProvider.CreateModelDisposer(compiledActionDescriptor);
             }
 
+            var pageStartFactories = GetPageStartFactories(compiledActionDescriptor);
+
             return new PageActionInvokerCacheEntry(
                 compiledActionDescriptor,
                 pageFactory,
                 pageDisposer,
                 modelFactory,
                 modelReleaser,
+                pageStartFactories,
```

Comment:
We are going to rely on the `IActionDescriptorChangeProvider` invalidating the cache anytime any cshtml file is invalidated. This would allow us to cache ViewStarts with the page's cache.